### PR TITLE
[FIX] Real names were not displayed in the reactions (API/UI)

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
Closes https://github.com/RocketChat/Rocket.Chat/issues/12889

### CODE REVIEW

**Comment by MarcosSpessatto** [comment]
@rodrigok done

File: packages/rocketchat-utils/server/lib/composeMessageObjectWithUser.js [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
 
 export const composeMessageObjectWithUser = function(message, userId) {
 	if (message) {
 		if (message.starred && Array.isArray(message.starred)) {
 			message.starred = message.starred.filter((star) => star._id === userId);
 		}
-		if (message.u && message.u._id && settings.get('UI_Use_Real_Name')) {
-			const user = getUser(message.u._id);
-			message.u.name = user && user.name;
+		if (message.u && message.u.name && !settings.get('UI_Use_Real_Name')) {
```
Comment (sampaiodiego) [inline packages/rocketchat-utils/server/lib/composeMessageObjectWithUser.js]: why did you change to remove the field instead of adding it?

Comment (MarcosSpessatto) [reply packages/rocketchat-utils/server/lib/composeMessageObjectWithUser.js]: In this case, because the real name is already saved in the `Messages` collection(`mentions` and `username`). So if you are not using `Real Names`, we remove the real name.

Comment (sampaiodiego) [reply packages/rocketchat-utils/server/lib/composeMessageObjectWithUser.js]: the problem is that property is not updated (like if an user changes its name), that's why we always inject it before sending to client. I don't even know why we save it there in the first place.

**Comment by rodrigok** [comment]
@MarcosSpessatto can you fix the conflicts?

**Comment by MarcosSpessatto** [comment]
@rodrigok done

File: packages/rocketchat-utils/server/lib/composeMessageObjectWithUser.js (hunk 1 above)
Comment (MarcosSpessatto) [reply packages/rocketchat-utils/server/lib/composeMessageObjectWithUser.js]: I changed to the old behavior.
